gcloud-java-dns/src/main/java/com/google/cloud/dns/testing/LocalDnsHelper.java

```diff
@@ -328,6 +361,62 @@ public void handle(HttpExchange exchange) throws IOException {
           requestMethod, exchange.getRequestURI())));
     }
 
+    private Response handleBatch(final HttpExchange exchange) throws IOException {
+      String contentType = exchange.getRequestHeaders().getFirst("Content-type");
+      if (contentType != null) {
+        int port = server.getAddress().getPort();
+        HttpMediaType httpMediaType = new HttpMediaType(contentType);
+        String boundary = httpMediaType.getParameter("boundary");
+        MultipartStream multipartStream =
+            new MultipartStream(exchange.getRequestBody(), boundary.getBytes(), 1024, null);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        byte[] bytes = new byte[1024];
+        multipartStream.skipPreamble();
+        while (multipartStream.readBoundary()) {
+          Socket socket = new Socket("localhost", port);
+          OutputStream socketOutput = socket.getOutputStream();
+          ByteArrayOutputStream section = new ByteArrayOutputStream();
+          multipartStream.readBodyData(section);
+          BufferedReader reader = new BufferedReader(
+              new InputStreamReader(new ByteArrayInputStream(section.toByteArray())));
+          String line;
+          String contentId = null;
+          while (!(line = reader.readLine()).isEmpty()) {
+            if (line.toLowerCase().startsWith("content-id")) {
+              contentId = line.split(":")[1].trim();
+            }
+          }
+          String requestLine = reader.readLine();
+          socketOutput.write((requestLine + "  \r\n").getBytes());
+          socketOutput.write("Connection: close \r\n".getBytes());
+          while ((line = reader.readLine()) != null) {
+            socketOutput.write(line.getBytes());
+            if (!line.isEmpty()) {
+              socketOutput.write(" \r\n".getBytes());
+            } else {
+              socketOutput.write("\r\n".getBytes());
+            }
+          }
+          socketOutput.flush();
+          InputStream in = socket.getInputStream();
+          int length;
+          out.write(responseSeparator.getBytes());
+          out.write("Content-Type: application/http \r\n".getBytes());
+          out.write(("Content-ID: " + contentId + " \r\n\r\n").getBytes());
+          try {
+            while ((length = in.read(bytes)) != -1) {
+              out.write(bytes, 0, length);
+            }
+          } catch (IOException ex) {
+            // this handles connection reset error
```

Comment:
I did not find another way of detecting the connection reset. In that case, any attempted read from the socket throws an exception instead of -1. Yet, available() returns true. Any suggestions?